tests/browser/test/test_setup.js

```diff
@@ -232,11 +232,19 @@ async function dragBlockTypeFromFlyout(browser, categoryName, type, x, y) {
 }
 
 async function contextMenuSelect(browser, block, itemText) {
-  await block.click({button: 2});
-  await browser.pause(200);
+  // Clicking will always happen in the middle of the block's bounds
+  // (including children) by default, which causes problems if it has holes
+  // (e.g. statement inputs).
+  const xOffset = Math.round((await block.getSize('width')) * 0.3);
+  const yOffset = Math.round((await block.getSize('height')) * 0.45);
+
+  await block.click({button: 2, x: -xOffset, y: -yOffset});
```

Comment:
I'm a bit perplexed about the values you've chosen here.  If I am reading correctly, you're clicking 30% in from the right and 45% down from the top, it looks like it's optimised to be in the middle of the opening of C-shaped block, at least if it has any blocks inside it.  Wouldn't, say, 5% in from the left and somewhere >50% down from the top be a better bet?  Or small fixed offsets from the upper-left corner (chosen carefully to avoid any icons)?

Reply:
The default click position is in the middle, so we're actually clicking 5% from the top and 20% from the left (this is why the offsets are negative). For some reason clicking closer to the left was being flakey, and sometimes clicked the workspace instead. Through experimentation, this seemed to work consistently!

I can expand on the comment to say where the click is occuring.